### PR TITLE
All/dp 21258 add brand banner

### PR DESCRIPTION
**Pull request by clairesunstudio**
Add BrandBanner in Patternlab and Core, to allow Mayflower users to implement
Design prototype: https://eotss.invisionapp.com/console/share/8K202Y5GUR/497139963

#### Added:
  - project: Patternlab, Core
    component: BrandBanner
    description: Add BrandBanner molecule in Patternlab and generate the HTML, CSS and JS for Mayflower core documentation. (#1362)
    issue: DP-21258
    impact: Minor
#### Changed:
  - project: React
    component: BrandBanner
    description: Change toggle rendering logic in React to match Patternlab. (#1362)
    issue: DP-21258
    impact: Patch

---

### Core
External review URL --> https://mayflower.digital.mass.gov/core/b/all/DP-21258-add-brand-banner/index.html?path=/docs/components-brandbanner--brand-banner-example

> Generated `brand-banner(.min).css` and `brand-banner(.min).js` in the mayflower-assets package, will be available via CDN after release. Locally they are in assets/build/css and assets/build/js after running `rush build`

Validate that the HTML, CSS and JS generated are working:
https://codepen.io/clairesun/pen/KKNQQLx

---

### Patternlab
External review URL --> https://mayflower.digital.mass.gov/patternlab/b/all/DP-21258-add-brand-banner/?p=molecules-brand-banner
brandbanner-light

by default:
```
{
  "brandBanner": {
    "bgTheme": "",
    "bgColor": "",
    "hasSeal": true,
    "hasToggle": true
  }
}
```
other config options, e.g.
```
{
  "brandBanner": {
    "bgTheme": "dark",
    "bgColor": "c-primary-alt",
    "hasSeal": true,
    "hasToggle": true
  }
}
```

---

Pattenlab issues noticed:
- JS are not modular, no way to export named function
    - For a onClick function, the way currently allows is to export a self-executing anonymous function and attach the event to HTML using a selector. 
    - This then gets compiled into the index-generated.js which then runs these selector queries on all pages, this impairs performance
    -Screen Shot 2021-02-25 at 12 25 06 PM
Screen Shot 2021-02-25 at 12 24 56 PM
Screen Shot 2021-02-25 at 2 54 21 PM

- Jquery dependency should be removed, at least for core components
- Icon 
    - No straight forward way to add and configure svgs
        - svg2twig no longer used but documentation and twigs templates are not cleaned up
        - svg4everybody is used to create svg mapping symbol
        - svg styles are not uniformed and hard to style and size

### CODE REVIEW

**Comment by clairesunstudio** [comment]
hmm I noticed that the changes are not live on the test URL for some reason... looking into it now https://mayflower.digital.mass.gov/core/b/all/DP-21258-add-brand-banner/index.html?path=/docs/components-brandbanner--brand-banner-example

**Comment by ygannett** [comment]
For the backstop failures in PATTERNLAB, it's from the relationship indicator.  You can find samples with information details pages. ex) http://localhost:3000/patterns/05-pages-information-details/05-pages-information-details.html

**develop**
develop

**This branch**
Mass_Gov

I tried to merge the latest `develop`, but it's already up to date. In the `develop`, the component looks/functions fine. So, it seems some change in this PR could be affecting it.

No change with the JS file to control the display of the relationship indicator (packages/patternlab/styleguide/source/assets/js/modules/headerTags.js) in this branch.

I tried to comment out the packages/patternlab/styleguide/source/assets/js/modules/brandBanner.js in index.js.  The relationship indicator was rendered fine.  There might be some conflicts.

**Comment by clairesunstudio** [comment]
@ygannett ya, I'm very confused about the relationship indicator visual regression.... still looking into it

**Comment by ygannett** [comment]
@clairesunstudio In the info details sample page, it shows these errors.  Hope this helps.
Mass_Gov_and_cd___MassIT_mayflower___nvm_use_12__rush_install__rush_start_patternlab

**Comment by clairesunstudio** [comment]
@ygannett thanks I think it's fixed 🤞

**Comment by clairesunstudio** [comment]
ah I also found out why the test URL doesn't get updated.. it's because of the cloudfront cache
Screen Shot 2021-02-26 at 5 04 31 PM

File: packages/patternlab/styleguide/source/_patterns/01-atoms/09-media/site-logo.twig [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <div class="ma__site-logo">
   <a href="/" title="Mass.gov home page">
-    <img src="{{ sealImage }}" alt="" width="45" height="45" />
+    <img src="{{ sealImage }}" alt="Massachusetts State Seal" width="45" height="45" />
```
Comment (ygannett) [inline packages/patternlab/styleguide/source/_patterns/01-atoms/09-media/site-logo.twig]: This alt is intentionally left blank for the Mass.gov site title in the header.  In this context, the component has the text content of "Mass.gov" as the site title and screen reader users wouldn't get much of helpful information from "Massachusetts State Seal."  So, the image is treated as decorative.

**Review by ygannett — COMMENTED** [review]
Please see my comment for the site-logo.twig.

File: packages/patternlab/styleguide/source/_patterns/01-atoms/09-media/site-logo.twig (hunk 1 above)
Comment (clairesunstudio) [inline packages/patternlab/styleguide/source/_patterns/01-atoms/09-media/site-logo.twig]:
```suggestion
    <img src="{{ sealImage }}" alt="" width="45" height="45" />

```

Comment (clairesunstudio) [reply packages/patternlab/styleguide/source/_patterns/01-atoms/09-media/site-logo.twig]: I see, removed

**Review by ygannett — APPROVED** [review]

**Pull request merged**